tensorflow/lite/kernels/strided_slice.cc

```diff
@@ -172,39 +218,62 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
     SetTensorToDynamic(op_context.output);
     return kTfLiteOk;
   }
-  return ResizeOutputTensor(context, &op_context);
+  if (op_context.begin->type == kTfLiteInt64) {
+    return ResizeOutputTensor<int64_t>(context, &op_context);
+  } else if (op_context.begin->type == kTfLiteInt32) {
+    return ResizeOutputTensor<int32_t>(context, &op_context);
+  }
+  context->ReportError(
+      context,
+      "Beign-End-Stride of type '%s' are not supported by StridedSlice.",
+      TfLiteTypeGetName(op_context.begin->type));
+  return kTfLiteError;
 }
 
 template <KernelType kernel_type>
 TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   StridedSliceContext op_context(context, node);
 
   if (IsDynamicTensor(op_context.output)) {
-    TF_LITE_ENSURE_OK(context, ResizeOutputTensor(context, &op_context));
+    if (op_context.begin->type == kTfLiteInt64) {
+      TF_LITE_ENSURE_OK(context,
+                        ResizeOutputTensor<int64_t>(context, &op_context));
+    } else if (op_context.begin->type == kTfLiteInt32) {
+      TF_LITE_ENSURE_OK(context,
+                        ResizeOutputTensor<int32_t>(context, &op_context));
+    }
   }
 
-  std::vector<int32_t> starts;
-  std::vector<int32_t> stops;
-  std::vector<int32_t> strides;
+  std::vector<int64_t> starts;
+  std::vector<int64_t> stops;
+  std::vector<int64_t> strides;
 
   for (int i = op_context.dims; i < kMaxDim; i++) {
```

Comment:
While you are at it, could you please reserve the ```vector```s to ```kMaxDim```, since we know all of them will get atleast that many elements put in.

Reply:
If we reserve the size, starts, stops and strides variable can't be initialized properly in the below logic, so following existing logic not to break.